SIPS/sip-28.md

```diff
@@ -0,0 +1,97 @@
+---
+sip: 28
+title: Background Events
+status: Draft
+author: Olaf Tomalka (@ritave)
+created: 2024-10-09
+---
+
+## Abstract
+
+A few terse sentences that are a technical summary of the proposal. Someone should be able to read this paragraph and understand the gist of this SIP.
+
+## Motivation
+
+The "why"
+
+## Specification
+
+> Indented sections like this are considered non-normative.
+
+Formal specification of the proposed changes in the SIP. The specification must be complete enough that an implementation can be created from it.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+This SIP introduces a new permission `endowment:background-event`. This permissions grants a Snap the ability to schedule future events.
+
+This permission takes no parameters and is specified in the `snap.manifest.json` as follows:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:background-event": {}
+  }
+}
+```
+
+### RPC Methods
+
+#### `snap_backgroundEventSchedule`
+
+This method allows a Snap to schedule a callback to `onBackgroundEvent` handler in the future with a JSON-RPC request object as a parameter.
+
+The RPC method takes two parameters:
+
+- `date` - An [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date and time and optional timezone offset.
```

Comment:
Is `date` preferred to letting the caller specify a number of seconds or something?

Wondering what is the simplest for developers.

Reply:
* Notice that we have minute rather than second level precision.
* Conversion to ISO strings is a single call to `date.toISOString()`.
* What happens to seconds when daylight time change happens?
* What happens to seconds when user travels between timezones (local-time vs UTC distinction is an edge case here, where local-time changes while UTC one doesn't)